BridgeCareApp/BridgeCare/DataAccessLayer/DataAdapters/AttributeYearDataTableAdapter.cs

```diff
@@ -0,0 +1,87 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Web;
+using System.Data;
+using System.Data.SqlClient;
+using BridgeCare.Models;
+
+namespace BridgeCare.DataAccessLayer.DataAdapters
+{
+    /// <summary>
+    /// Populate list of AttributeByYearModel
+    /// </summary>
+    public class AttributeYearDataTable
+    {
+        private DataTable AttributesDataTable;
+        private List<AttributeByYearModel> TheAttributeYearList;
+
+        public AttributeYearDataTable(DataTable tableWithAttributes)
+        {
+            AttributesDataTable = tableWithAttributes ?? throw new ArgumentNullException(nameof(tableWithAttributes));
+
+            TheAttributeYearList =  DataTableToAttributeList(tableWithAttributes);
```

Comment:
TheAttributeList - > AttributeValuesByYear

Reply:
this code was moved back into the AttributesByYear data access layer class. At the end of the conversation we realized it was kind of custom code for just attribute tables. It could stand alone as if no columns names *_YYYY existed it would just return empty, not crash but it is so particular to the 2 queries on attributes it should go back into the class.